dashboard/app/controllers/level_starter_assets_controller.rb

```diff
@@ -1,5 +1,6 @@
 class LevelStarterAssetsController < ApplicationController
   before_action :set_level
+  skip_before_action :verify_authenticity_token, only: [:destroy]
```

Comment:
remove hopefully unneeded code? If this breaks functionality when the delete is issued from within the asset manager in code studio (as opposed to using curl, storybook, etc), let's try debugging a bit further to hopefully get rid of this and close up any XSS vulnerability.

Reply:
it does break the functionality, and i'm stuck on debugging. removing this line means we need to supply the authenticity token from the client, which i've added, but the endpoint still raises an `ActionController::InvalidAuthenticityToken` error. these are the request headers:

```
DELETE /level_starter_assets/U5%20Lists%20Investigate%20Outfitly%20Code/1_goal.mp3 HTTP/1.1
Host: localhost-studio.code.org:3000
Connection: keep-alive
Pragma: no-cache
Cache-Control: no-cache
Accept: */*
X-CSRF-Token: +AODJDwvW8squAqbEu92mTpvckeu7iMxND1bL98jjVSKAN3wtich+2bTVRzl5FOvilZNEM23Tf7nwnGaPURRqw==
X-Requested-With: XMLHttpRequest
User-Agent: Mozilla/5.0 (Macintosh; Intel Mac OS X 10_14_5) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/76.0.3809.100 Safari/537.36
Content-Type: application/json; charset=utf-8
Origin: http://localhost-studio.code.org:3000
Referer: http://localhost-studio.code.org:3000/levels/16718/edit_blocks/start_blocks
Accept-Encoding: gzip, deflate
Accept-Language: en-US,en;q=0.9
Cookie: _ga=GA1.1.139698532.1540254904; optimizelyEndUserId=oeu1553717844596r0.07535817422503488; _ga=GA1.2.139698532.1540254904; storage_id_development=%2507%25C3%251E%2505P%2510dg%25A7%25D8%250DY%258B9Y%2595; _gid=GA1.1.370944268.1566183895; _gid=GA1.2.1492567763.1566240734; _limit_project_types_development=false; _user_type_development=teacher; _shortName_development=Dev; _experiments_development=%5B%5D; _gat=1; _learn_session_development=Vi8vOWovZEtsTFI4OUZNWHNpbkZSTHJFMTUvNnpkZit5aThMTFJYY1NBYzh1eUt1OVBDTnFyRzBPVzJ4OWtyU3hQN0EvOWcwTU5VRk5aZXAxeHRlTDE5UzFOUEpDVDR2WHllMHlCMjhOekdzUWFtZ1VXM3hGZ04rbnpTQmNPWEJaU1dpKzNUVmZNS0Y2T29sNkdWNzRtbXF5c1FiM1pQRGFqbUZ2ZVpiZ0hsbzVxSUNNdEZmSmpZTWRpV21WY0JQMUgvQ2tLaSsrYXFpeVRmaFFWNWN4NGQyWDNZbFJhVzZUWWVIbGxaN0J4a05yMm9wU0R1MlkvMDZleFBheGJ0dEZhVFlQbS94R1ZQalUrNHdaaWVnQ3c9PS0tVEFYOU9zQUNDUEh3UHEybGRWRysrQT09--5344e079ff1dbc6a033424512bb868ea290abba4
```

do you notice something i'm missing here?

Reply:
I don't have a diagnosis, but here are some leads:

1. the authenticity token could be stale: https://stackoverflow.com/questions/3364492/actioncontrollerinvalidauthenticitytoken (although the url you're hitting shouldn't be cached, so maybe not)

2. there might be a problem with authenticating the user before checking for forgery: https://stackoverflow.com/questions/38331496/rails-5-actioncontrollerinvalidauthenticitytoken-error/39954005 you might be able to fix this with `before_action :authenticate_user!` https://stackoverflow.com/questions/38331496/rails-5-actioncontrollerinvalidauthenticitytoken-error/39954005 .  or this might go away when you merge in your cancan changes from the previous PR

Reply:
From our team meeting, going to put implementing this on hold 😞 and have logged in Jira ([LP-714](https://codedotorg.atlassian.net/browse/LP-714))

Reply:
thanks Maddie, please let me know when you are ready for another look

Reply:
ready when you are!